panic: runtime error: invalid memory address or nil pointer dereference at (*stream).isMigrating

Observing panic while starting jetstream in 3rd(last) node of js cluster

```

[1] 2022/07/21 04:42:32.138888 [[36mDBG[0m] Exiting stream monitor for '$G > DUMMY' [S-R3F-F3VzVqFJ]
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x80e7f8]

goroutine 77 [running]:
github.com/nats-io/nats-server/v2/server.(*stream).isMigrating(0xc00048de18)
	/home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_cluster.go:1937 +0x38
github.com/nats-io/nats-server/v2/server.(*jetStream).monitorStream(0xc00012a370, 0x0, 0xc00007e3f0, 0x0)
	/home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_cluster.go:1773 +0xf0c
github.com/nats-io/nats-server/v2/server.(*jetStream).processClusterCreateStream.func1()
	/home/travis/gopath/src/github.com/nats-io/nats-server/server/jetstream_cluster.go:2765 +0x2a
created by github.com/nats-io/nats-server/v2/server.(*Server).startGoRoutine
	/home/travis/gopath/src/github.com/nats-io/nats-server/server/server.go:3016 +0x87
```

Attached -DV logs
[nats.log](https://github.com/nats-io/nats-server/files/9156234/nats.log)

**Versions of nats-server and affected client libraries used:**
nats-server: 2.8.4
nats.go: 1.16.0


**OS/Container environment:**
docker
linux

--- a/server/jetstream_cluster.go
+++ b/server/jetstream_cluster.go
@@ -2050,6 +2050,10 @@ func (js *jetStream) monitorStream(mset *stream, sa *streamAssignment, sendSnaps
 
 // Determine if we are migrating
 func (mset *stream) isMigrating() bool {
+	if mset == nil {
+		return false
+	}
+
 	mset.mu.RLock()
 	js, sa := mset.js, mset.sa
 	mset.mu.RUnlock()

